docs/storage.md

```diff
@@ -468,6 +469,81 @@ config:
+The default HTTP Client comes with this Thanos OCI Object Storage should be able to handle most of the use cases.
+
+By default, like other clients, Thanos OCI Object Storage client will store the data from a single prometheus instance to the top directory of the bucket. 
+This means each bucket can only store one prometheus instance data.
+
+To support multiple prometheus instances with a single bucket, a new `object_base_path` option is introduced in OCI Object Storage client. 
+If defined, the value will be used as the prefix of the prometheus data in OCI bucket.
+
+For example, if you want to use the same bucket for 2 prometheus instances, you can define `object_base_path` as `/path/for/prometheus1` 
+for the first prometheus instance, and `/path/for/prometheus2` for the second prometheus instance.
+
+#### Default Provider
+The default config provider will look for configurations in 3 places: file in `$HOME/.oci/config`, `$HOME/.obmcs/config`, and
+variables names starting with the string TF_VAR. If the same configuration is found in multiple places the provider will
+prefer the first one.
+
+For Example:
+```yaml
+type: OCI
+config:
+  provider: "default"
+  bucket: ""
+  compartment_ocid: ""
+  object_base_path: ""            // Optional object base path (prefix) in OCI bucket.
+  part_size: ""                   // Optional part size to override the OCI default of 128 MiB, value is in bytes.
+  max_request_retries: ""         // Optional maximum number of retries for a request.
+  request_retry_interval: ""      // Optional sleep duration in seconds between retry requests.
+  http_config:
+    idle_conn_timeout: 1m30s      // Optional maximum amount of time an idle (keep-alive) connection will remain idle before closing itself. Zero means no limit.
+    response_header_timeout: 2m   // Optional amount of time to wait for a server's response headers after fully writing the request.
+    tls_handshake_timeout: 10s    // Optional maximum amount of time waiting to wait for a TLS handshake. Zero means no timeout.
+    expect_continue_timeout: 1s   // Optional amount of time to wait for a server's first response headers. Zero means no timeout and causes the body to be sent immediately.
+    insecure_skip_verify: false   // Optional. If true, crypto/tls accepts any certificate presented by the server and any host name in that certificate.
+    max_idle_conns: 100           // Optional maximum number of idle (keep-alive) connections across all hosts. Zero means no limit.
+    max_idle_conns_per_host: 100  // Optional maximum idle (keep-alive) connections to keep per-host. If zero, DefaultMaxIdleConnsPerHost=2 is used.
+    max_conns_per_host: 0         // Optional maximum total number of connections per host.
+    disable_compression: false    // Optional. If true, prevents the Transport from requesting compression.
+```
+
+
+#### Instance Principle Provider
```

Comment:
This should be _Principal_ as per the `provider` value below. :)

Reply:
Same, pushed the fix a minute ago.